### PR TITLE
Update handling of FIFE exceptions

### PR DESCRIPTION
**Pull request by Niektory**
Issue #2583

### CODE REVIEW

File: horizons/gui/tabs/boatbuildertabs.py [hunk 1]
```diff
@@ -163,14 +164,10 @@ def update_queue(self, container_active):
 
 			try:
 				icon = Icon(name=icon_name, image=image, helptext=helptext)
-			except RuntimeError,e:
+			except fife.NotFound,e:
```
Comment (MasterofJOKers) [inline horizons/gui/tabs/boatbuildertabs.py]: I'd really like this to change to `fife.NotFound as e` or *at least* add a space between the ',' and the 'e'.

File: run_uh.py [hunk 2]
```diff
@@ -355,6 +355,12 @@ def setup_fife():
 	else:
 		log().debug('Using fife version %s.%s.%s, at least %d.%d.%d required', fife_version_major, fife_version_minor, fife_version_patch, VERSION.REQUIRED_FIFE_MAJOR_VERSION, VERSION.REQUIRED_FIFE_MINOR_VERSION, VERSION.REQUIRED_FIFE_PATCH_VERSION)
 
+	# FIFE 0.4.0 changed the types of exceptions from RuntimeError to FIFE-specific types
+	# this line is needed for old FIFE exception handlers to work
+	# when all handlers are updated to use the new types it can be removed
+	# see issue #2583
+	fife.Exception.__bases__ = RuntimeError,
```
Comment (MasterofJOKers) [inline run_uh.py]: This looks really very hacky. Are there that many exception handlers left or what's the problem with changing the remaining handlers?

**Comment by MasterofJOKers** [comment]
All the GUI tests in travis failed. That seems to be because of the older fife version available to travis. What's interesting is, that it does not fail with "we need a newer version of fife", which is bad.

**Comment by LinuxDonald** [comment]
Here are all runetime errors in the code:
https://github.com/unknown-horizons/unknown-horizons/search?p=2&q=RuntimeError&utf8=%E2%9C%93

And here are the new typs that must be used:
https://github.com/fifengine/fifengine/commit/4c34e521d4160a9b5ea7890586404bdcacd9b310

**Comment by MasterofJOKers** [comment]
I can do a grep, but thanks for the link to the exception types.
Still, I wanted to know, why he didn't convert the other few. My guess would be: because we don't know what exact error to fetch. @Niektory Is there a base class for all exceptions fife raises? Is fife.Exception that class? Why couldn't we use that one instead of the workaround, where we are not sure?

**Comment by MasterofJOKers** [comment]
Oh right, the git version of fife we are using already says its 0.4.0, so that's why the tests don't fail earlier.

**Comment by Niektory** [comment]
I went for minimal changes to avoid breaking anything. But maybe I was too conservative. I'll try updating all of them, then.
Yes, fife.Exception is the base class for all FIFE exceptions.

**Comment by Niektory** [comment]
OK, done. I think this should do the trick.
I'm not sure about e137a4b. I assumed that code was supposed to handle the case of using an unknown font, which throws an `InitializationError` from Pychan (which isn't one of the new exception types so it seems it was already wrong before the recent FIFE change). So don't pull that commit if you believe I'm mistaken.

**Comment by CrafterSvK** [comment]
Main branch doesn't work for me. This one does. thanks :)

**Comment by LinuxDonald** [comment]
We have an other problem there too see line 114: http://pastebin.com/iPc2DEwK

**Comment by Niektory** [comment]
@LinuxDonald That seems to be caused by #2516. I'll comment there.

**Pull request merged**